multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -690,61 +727,35 @@ def infer_problem_type_output_shape(
         Shape of output.
     """
     if label_column is None:
-        return provided_problem_type, None
-
-    if provided_problem_type is not None:
-        if provided_problem_type == MULTICLASS or provided_problem_type == BINARY:
-            class_num = len(data[label_column].unique())
-            err_msg = (
-                f"Provided problem type is '{provided_problem_type}' while the number of "
-                f"unique values in the label column is {class_num}."
-            )
-            if provided_problem_type == BINARY and class_num != 2:
+        return None
+
+    if problem_type in [BINARY, MULTICLASS, REGRESSION]:
+        class_num = len(data[label_column].unique())
+        err_msg = (
+            f"Problem type is '{problem_type}' while the number of "
+            f"unique values in the label column is {class_num}."
+        )
+        if problem_type == BINARY:
+            if class_num != 2:
                 raise AssertionError(err_msg)
-            elif provided_problem_type == MULTICLASS and class_num <= 2:
+            return 2
+        elif problem_type == MULTICLASS:
+            if class_num <= 2:
                 raise AssertionError(err_msg)
-            return provided_problem_type, class_num
-        if provided_problem_type == BINARY:
-            return BINARY, 2
-        elif provided_problem_type == MULTICLASS:
-            class_num = len(data[label_column].value_counts())
-            return MULTICLASS, class_num
-        elif provided_problem_type == CLASSIFICATION:
-            class_num = len(data[label_column].value_counts())
-            if class_num == 2:
-                return BINARY, 2
-            else:
-                return MULTICLASS, class_num
-        elif provided_problem_type == REGRESSION:
-            return provided_problem_type, 1
-        elif provided_problem_type == NER:
-            return provided_problem_type, None
-        elif provided_problem_type == OBJECT_DETECTION:
-            return provided_problem_type, None
-        elif provided_problem_type == OPEN_VOCABULARY_OBJECT_DETECTION:
-            return provided_problem_type, None
-        else:
-            raise ValueError(
-                f"Problem type '{provided_problem_type}' doesn't have a valid output shape "
-                f"for training. The supported problem types are"
-                f" '{BINARY}', '{MULTICLASS}', '{REGRESSION}',"
-                f" '{CLASSIFICATION}', '{NER}',"
-                f" '{OBJECT_DETECTION}', '{OPEN_VOCABULARY_OBJECT_DETECTION}'"
-            )
+            return class_num
+        elif problem_type == REGRESSION:
+            if class_num <= 1:  # in case users want to try toy datasets
```

Comment:
Is there a reason to raise error when `class_num==1`? I think it's fine if user want a sanity test with only one data row.

Reply:
Updated to `class_num < 1`.